// react-pie3d/dist/elements/path.test.js
"use strict";
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
var react_1 = __importDefault(require("react"));
var react_test_renderer_1 = require("react-test-renderer");
var const_1 = require("../const");
var types_1 = require("../types");
var path_1 = require("./path");
describe('Path component tests', function () {
    test('it render proper path', function () {
        var data = {
            color: 'mediumvioletred',
            endAngle: 3.141592653589793,
            index: 1,
            moved: false,
            percentageValue: 0.3333333333333333,
            startAngle: 1.0471975511965976,
            value: 20,
        };
        var pathVariables = __assign(__assign({}, const_1.DEFAULT_CONFIG), { moveElement: function () { return null; }, rx: 150, ry: 100 });
        var path = react_test_renderer_1.create(react_1.default.createElement(path_1.Path, { data: data, pathVariables: pathVariables, type: types_1.PathType.Top }));
        var instance = path.root;
        var titles = instance.findAllByType('title');
        expect(titles[0].children[0]).toBe('20 33.33%');
        expect(titles.length).toBe(1);
    });
});
